src/app/shared/pricing-definitions/pricing-definition/pricing-definition.component.html

```diff
@@ -252,6 +252,132 @@
+                    <mat-slide-toggle (change)="toggle($event)" id="maxEnergyKWh" [checked]="maxEnergyKWhEnabled.value">
+                      {{'settings.pricing.restriction_max_energy' | translate}}
+                    </mat-slide-toggle>
+                  </div>
+                  <!-- value max energy -->
+                  <div class="col-md-2">
+                    <mat-form-field *ngIf="maxEnergyKWhEnabled.value">
+                      <mat-label>{{'settings.pricing.restriction_max_energy_unit' | translate}}</mat-label>
+                      <input matInput type="text" placeholder="{{'settings.pricing.restriction_max_energy_unit' | translate}}"
+                      [formControl]="maxEnergyKWhValue" required/>
+                      <mat-error *ngIf="maxEnergyKWhValue?.errors?.required">{{'general.mandatory_field' | translate}}</mat-error>
+                      <mat-error *ngIf="maxEnergyKWhValue?.errors?.pattern">{{'general.error_number_pattern' | translate}}</mat-error>
+                    </mat-form-field>
+                  </div>
+                </div>
+                <!-- time range -->
+                <div class="row">
+                  <div class="col-md-2 pricing-dimension">
+                    <mat-slide-toggle (change)="toggle($event)" id="timeFrom" [checked]="timeFromEnabled.value">
+                      {{'settings.pricing.restriction_time_range' | translate}}
+                    </mat-slide-toggle>
+                  </div>
+                  <!-- start time -->
+                  <div class="col-md-2">
+                    <mat-form-field *ngIf="timeFromEnabled.value">
+                      <mat-label>{{'settings.pricing.restriction_start_time' | translate}}</mat-label>
+                      <input matInput type="time" id="timeFromValue" name="timeFromValue" [formControl]="timeFromValue" (change)="setMinTime($event)" required/>
+                    </mat-form-field>
+                    <mat-error *ngIf="timeFromValue?.errors?.required">{{'general.mandatory_field' | translate}}</mat-error>
+                    <mat-error *ngIf='timeToValue?.errors?.incorrectEndTime'>{{'settings.pricing.pricing_definition_time_error' | translate}}</mat-error>
+                  </div>
+                  <!-- end time -->
+                  <div class="col-md-2">
+                    <mat-form-field *ngIf="timeFromEnabled.value">
+                      <mat-label>{{'settings.pricing.restriction_end_time' | translate}}</mat-label>
+                      <input matInput type="time" id="timeToValue" name="timeToValue" [formControl]="timeToValue" required/>
+                    </mat-form-field>
+                    <mat-error *ngIf="timeToValue?.errors?.required">{{'general.mandatory_field' | translate}}</mat-error>
+                    <mat-error *ngIf='timeToValue?.errors?.incorrectEndTime'>{{'settings.pricing.pricing_definition_time_error' | translate}}</mat-error>
```

Comment:
There is no wrong end time now as it spans to the next day.

Reply:
you're right, changed it to 'timeRangeError' to display "same from/end time" error (if timeFromValue === timeEndValue)